airbyte-to-flow/src/interceptors/fix_document_schema.rs

```diff
@@ -188,12 +190,56 @@ pub fn remove_enums(schema: &mut serde_json::Value) {
     traverse_jsonschema(
         schema,
         &mut |map: &mut Map<String, serde_json::Value>, _| {
-            map.remove("enum");
+            if let Some(values) = map.remove("enum") {
+                // If the schema doesn't specify the type, then we'll add one to replace the enum we removed.
+                if !map.contains_key("type") {
+                    // Collect the json type of each enum value into a set.
+                    // Technically, one could have specified `enum: []`, in
+                    // which case we won't be able to set the type. Use of
+                    // `BTreeSet` here is an easy way to ensure consistent
+                    // ordering of types when there's multiple.
+                    let value_types = values.as_array().map(|vals| {
+                        vals.iter()
+                            .map(type_for)
+                            .collect::<std::collections::BTreeSet<_>>()
+                    });
+                    if let Some(new_types) = value_types {
+                        // Use a plain string value in the common case where all
+                        // enum values have the same type. Otherwise, type will
+                        // be an array.
```

Comment:
Supernit: I don't think this branching is really necessary, `type: ["string"]` is the same as `type: "string"`, so might simplify the code for us to always output an array, but it's not a strong opinion.

Reply:
yea, I went back and forth a bit, but I felt like the single type would be the more common case, and that using a string instead of an array might help make it slightly easier to read. IMO the branch is worth it here.